### ISSUE TITLE
 type cast in tuples

### ISSUE THREAD

**Issue by midiusRed** [Microsoft/TypeScript#28700]
**TypeScript Version:**  3.1.6

**Code**

```ts
		const a1:[number, number, boolean] = [1, 0, false];
		//... code
		if (a1[2]) {
			a1[2] = false; //error TS2322
		}
```

**Expected behavior:**
no error (run in TypeScript 2.9.2)

**Actual behavior:**
TS2322: Type 'false' is not assignable to type 'true'.

why is the type cast to true ?

**Comment by AlCalzone** [comment]
`if (a1[2])` checks for truthiness and thus narrows the type of `a[2]` from `boolean` to `true`.

**Comment by ajafff** [comment]
first of all, this has nothing to do with type casting. this is an issue with narrowing.

This is fixed by #27574 which is part of v3.2.0-rc (and probably this week in v3.2.1)
